### PR TITLE
Wasiu-hub-code-review

### PR DESCRIPTION
**Pull request by mike-shields-dev**
Hi Wasiu, nice work! Your code is very well formatted with succinct and easy-to-read solutions.

Consider using more "atomic" commits rather than one single commit. Here's an [article](https://dev.to/samuelfaure/how-atomic-git-commits-dramatically-increased-my-productivity-and-will-increase-yours-too-4a84#:~:text=By%20writing%20atomic%20git%20commits,already%20increases%20your%20productivity%20tremendously.) that you might find interesting.

There are some (minor) cases where you have slightly misunderstood the problems described in the track material, see the comments I've made. 

Going forward try to consider how robust your tests are:

- Are the tests too simple?
- Are there any "edge cases" that have not been covered?
- How should a function behave if not passed the expected argument/s?

Overall this is very good work, well done! :rocket: :crescent_moon:

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<!DOCTYPE html>
```
Comment (mike-shields-dev) [inline index.html:1]: I'm not sure why you have a html file in this project... :thinking:

File: src/kata1.fizzBuzz.js [hunk 2]
```diff
@@ -1,3 +1,31 @@
...
+//   if (number % 5 === 0) {
+//     return "Buzz";
+//   }
+//   if (number % 3 === 0) {
+//     return "Fizz";
+//   }
+//   return number;
+// };
 
+const fizzBuzz = (number) => {
```
Comment (mike-shields-dev) [inline src/kata1.fizzBuzz.js:14]: Nice! I like that you have taken the time to create the `isDivisible` variables to make the code easier to understand.

File: src/kata2.booleanToWord.js [hunk 3]
```diff
@@ -1,3 +1,8 @@
-const booleanToWord = (boolean) => {};
+const booleanToWord = (boolean) => {
```
Comment (mike-shields-dev) [inline src/kata2.booleanToWord.js:1]: This does pass the test, but the test requirements are not quite right as it tests for the **strings** `"true"` and `"false"` rather than the **booleans** `true` and `false`. 

The strings `"true"` and `"false"` are both **truthy** values. This has added complexity to this function by needing to conditionally check for the string `"true"`.

Testing for booleans (`true` or `false`), would have enabled a simpler function, for example, you could use the ternary operator:

```js
const booleanToWord = boolean => boolean ? "Yes" : "No"
```

File: test/kata2.booleanToWord.test.js [hunk 4]
```diff
@@ -2,4 +2,8 @@ const { booleanToWord } = require("../src");
 
 describe("booleanToWord", () => {
   // how do we create specs again???
+  test("returns the boolean word passed", () => {
```
Comment (mike-shields-dev) [inline test/kata2.booleanToWord.test.js:5]: See comment [here](https://github.com/mike-shields-dev/command-shift-diy-kata-code-reviews/pull/4/commits/6385ca8530ed4882477a33c335c2720cd6c7c3ec#diff-b3177de8bf616cf47ffdeb7fb107b2d75138137d9e297ed28b273361ca81c341)

File: test/kata3.numberToReversedDigits.test.js [hunk 5]
```diff
@@ -1,5 +1,8 @@
 const { numberToReversedDigits } = require("../src");
 
 describe("numberToReversedDigits", () => {
-  test("returns a reversed array of the number's digits", () => {});
-});
\ No newline at end of file
+  test("returns a reversed array of the number's digits", () => {
```
Comment (mike-shields-dev) [inline test/kata3.numberToReversedDigits.test.js:4]: The test requirements are not quite correct. It should be asserting that the function returns an array of numbers when passed an argument with the `number` datatype, not a `string`. 

```js
expect(numberToReversedDigits(12345)).toEqual([5, 4, 3, 2, 1])
```

This would have required a more challenging solution by performing the following operations: 
1. Convert the `number` argument to a string
2. Convert the string to an array
3. Reverse the array

File: test/kata4.humanCatDogYears.test.js [hunk 6]
```diff
@@ -1,3 +1,7 @@
 const { humanCatDogYears } = require("../src");
-
 // Look Ma, no handlebars!!!
+describe("humanCatDogYears", () => {
```
Comment (mike-shields-dev) [inline test/kata4.humanCatDogYears.test.js:3]: This test suite could be a little more robust by testing for all possible conditional branches. 

Currently, this test only asserts the behaviour of the function when the number passed in is `10` (a number that is not `1` or `2`). What about when the numbers `1` and `2` are passed as arguments? 

Someone else could change the logic for those conditions and the test suite would still pass.

File: test/kata5.reachDestination.test.js [hunk 7]
```diff
@@ -1,5 +1,7 @@
 const { reachDestination } = require("../src");
 
 describe("reachDestination", () => {
-  test("returns string with estimated time of arrival", () => {});
+  test("returns string with estimated time of arrival", () => {
```
Comment (mike-shields-dev) [inline test/kata5.reachDestination.test.js:4]: Consider whether it is a good idea to add more assertions to a test for extra assurance. 

For example, I could go and rewrite the `reachDestination` function to always return "I will be there in 4.5 hours" and this test would pass but it would not have the intended behaviour. 

I know that seems a bit pedantic in this case but by just adding at least one more assertion, you have made this test that little bit more robust.

File: test/kata6.joinNames.test.js [hunk 8]
```diff
@@ -1,5 +1,9 @@
 const { joinNames } = require("../src");
 
 describe("joinNames", () => {
-  test("returns string of names, seperated by commas and an ampersand", () => {});
+  test("returns string of names, seperated by commas and an ampersand", () => {
```
Comment (mike-shields-dev) [inline test/kata6.joinNames.test.js:4]: Again, similar to what is mentioned [here](https://github.com/mike-shields-dev/command-shift-diy-kata-code-reviews/pull/4/commits/6385ca8530ed4882477a33c335c2720cd6c7c3ec#r1258115593), consider adding more assertions for robustness. 

Something else to consider is how does the `joinNames` function behave when passed an array of name objects with a length < 3 ? :thinking:

**Review by mike-shields-dev — COMMENTED** [review]
Hi Wasiu, nice work! Your code is very well formatted with succinct and easy-to-read solutions. 

Consider using more "atomic" commits rather than one single commit. Here's an [article](https://dev.to/samuelfaure/how-atomic-git-commits-dramatically-increased-my-productivity-and-will-increase-yours-too-4a84#:~:text=By%20writing%20atomic%20git%20commits,already%20increases%20your%20productivity%20tremendously.) that you might find interesting. 

There are some (minor) cases where you have slightly misunderstood the problems described in the track material, see the comments I've made. Going forward try to consider how robust your tests are: 

- Are the tests too simple?
- Are there any "edge cases" that have not been covered?
- How should a function behave if not passed the expected argument/s?

Overall this is very good work, well done!. :rocket: :crescent_moon:

File: index.html (hunk 1 above)
Comment (Wasiu-hub) [reply index.html:1]: I was coding on VScode while I link the Html with JavaScript file to open the console in the browser
